responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -168,16 +168,59 @@ def __init__(self, model: Optional[Any], train: pd.DataFrame,
         if model is not None:
             # Cache predictions of the model
             self.predict_output = model.predict(
-                test.drop(columns=[target_column]))
+                self.get_test_data(
+                    test_data=test).drop(columns=[target_column]))
             if hasattr(model, SKLearn.PREDICT_PROBA):
                 self.predict_proba_output = model.predict_proba(
-                    test.drop(columns=[target_column]))
+                    self.get_test_data(
+                        test_data=test).drop(columns=[target_column]))
             else:
                 self.predict_proba_output = None
         else:
             self.predict_output = None
             self.predict_proba_output = None
 
+    def get_train_data(self):
+        """Returns the training dataset after dropping
+        features if such a list is configured.
+
+        :return: The training dataset after dropping features.
+        :rtype: pandas.DataFrame
+        """
+        if self._feature_metadata is None:
+            return self.train
+        else:
+            if self._feature_metadata.dropped_features is None or \
```

Comment:
I actually like the way we're using feature metadata as a container for things like dropped features. I'm just surprised we're not putting categorical features, and perhaps even target column into that container. I remember discussing this with @gaugup way back in June (?). Is there any particular reason we don't move them into feature metadata as well? Perhaps because it requires API changes?

Reply:
I know that categorical features is already in featureMetadata, and there is just a duplicate categorical_features in raiInsights that haven't been removed. Let's discuss this with Gaurav, maybe we can create a task for it so we can address this soon.